src/views/tables/PositionsTable.tsx

```diff
@@ -360,31 +359,36 @@ export const PositionsTable = ({
   const perpetualMarkets = useSelector(getPerpetualMarkets, shallowEqual) || {};
   const assets = useSelector(getAssets, shallowEqual) || {};
   const openPositions = useSelector(getExistingOpenPositions, shallowEqual) || [];
-  const openOrders = useSelector(getSubaccountOpenOrders, shallowEqual) || [];
-
-  const stopLossOrders: SubaccountOrder[] = [];
-  const takeProfitOrders: SubaccountOrder[] = [];
-
-  openOrders.map((order: SubaccountOrder) => {
-    if (isStopLossOrder(order)) {
-      stopLossOrders.push(order);
-    } else if (isTakeProfitOrder(order)) {
-      takeProfitOrders.push(order);
+  const { stopLossOrders: allStopLossOrders, takeProfitOrders: allTakeProfitOrders } = useSelector(
+    getSubaccountConditionalOrders,
+    {
+      equalityFn: (oldVal, newVal) => {
+        return (
+          shallowEqual(oldVal.stopLossOrders, newVal.stopLossOrders) &&
+          shallowEqual(oldVal.takeProfitOrders, newVal.takeProfitOrders)
```

Comment:
hmm ok I think this fixes it - though alternatively I could also write two selectors (one for SL, one for TP) and call them with the `shallowEqual` check. Thoughts here on what would be better? @aforaleka

Reply:
works for me!

Reply:
ah you're right - I think I noticed that too earlier; it seems sorta laggy - lemme re-try here 😢